manifests/repo/apt/cloudsmith.pp

```diff
@@ -0,0 +1,30 @@
+# erlang cloudsmith apt repo
+class erlang::repo::apt::cloudsmith (
+  String $ensure      = $erlang::repo::apt::ensure,
+  String $location    = 'https://dl.cloudsmith.io/public/rabbitmq/rabbitmq-erlang/deb/debian',
+  # trusty, xenial, bionic, etc
+  String $release     = downcase($facts['os']['distro']['codename']),
+  String $repos       = 'main',
+  String $key         = 'A16A42516F6A691BC1FF5621E495BB49CC4BBE5B',
+  String $key_source  = 'https://dl.cloudsmith.io/public/rabbitmq/rabbitmq-erlang/gpg.E495BB49CC4BBE5B.key',
+  Optional[Variant[Numeric, String]] $pin = $erlang::package_apt_pin,
```

Comment:
Do docstrings for these variables make sense?  If there is room for something like:

```puppet
class { 'erlang':
  ...
}
class { 'erlang::repo::apt::cloudsmith':
 ...
}
```
with parameters in both clases, then maybe yes, and in this case you can see the beginning of `init.pp` to see how to add docstrings . If the above does not make sense, then docstrings are probably not useful, and some refactoring could be done to fix this weirdness ;-)

Reply:
fixed the main weirdness for cloudsmith repository. 
Refactoring of the remaining things should most likly be in a separate PR to not mix context.